packages/smooth_app/lib/temp/product_preferences.dart

```diff
@@ -12,12 +12,57 @@ import 'package:openfoodfacts/model/AttributeGroup.dart';
 import 'package:openfoodfacts/model/Product.dart';
 
 // Project imports:
-import 'package:smooth_app/temp/user_preferences.dart';
 import 'package:smooth_app/temp/preference_importance.dart';
 
-class UserPreferencesModel extends ChangeNotifier {
-  UserPreferencesModel._();
+/// Where we store all product attribute groups and importance.
+///
+/// At init time we need to load from json the localized reference data
+/// for importance and attribute groups.
+/// We also need to store the app or user preferences regarding importance,
+/// e.g. 'nutriscore' is 'very_important' and 'ecoscore' is 'mandatory',
+/// but we cannot decide here how to store those preferences and rather let
+/// the developer implement [saveImportance] and [loadImportance]
+class ProductPreferences extends ChangeNotifier {
```

Comment:
Just checking: the extends ChangeNotifier is not going to create issues for apps that don't use Provider?

Reply:
I have to double-check that. Anyway we may not need to extend `ChangeNotifier`, but add a `notifyListeners` parameter in the constructor.